pypesto/sample/variational_inference.py

```diff
@@ -0,0 +1,153 @@
+        Number of iterations for the optimization.
+    method: str or :class:`Inference`
+        string name is case-insensitive in:
+            -   'advi'  for ADVI
+            -   'fullrank_advi'  for FullRankADVI
+            -   'svgd'  for Stein Variational Gradient Descent
+            -   'asvgd'  for Amortized Stein Variational Gradient Descent
+    n_samples:
+        Number of samples to generate after optimization.
+    random_seed: int
+        random seed for reproducibility
+    start_sigma: `dict[str, np.ndarray]`
+        starting standard deviation for inference, only available for method 'advi'
+    x0:
+        Initial parameter for the variational optimization. If None, the best parameter
+        found in optimization is used.
+    result:
+        A result to write to. If None provided, one is created from the
+        problem.
+    filename:
+        Name of the hdf5 file, where the result will be saved. Default is
+        None, which deactivates automatic saving. If set to
+        "Auto" it will automatically generate a file named
+        `year_month_day_profiling_result.hdf5`.
+        Optionally a method, see docs for `pypesto.store.auto.autosave`.
+    overwrite:
+        Whether to overwrite `result/sampling` in the autosave file
+        if it already exists.
+
+    Returns
+    -------
+    result:
+        A result with filled in sample_options part.
+    """
+    # prepare result object
+    if result is None:
+        result = Result(problem)
+
+    # number of samples
+    if n_iterations is not None:
+        n_iterations = bound_n_samples_from_env(n_iterations)
+
+    # try to find initial parameters
+    if x0 is None:
+        result.optimize_result.sort()
+        if len(result.optimize_result.list) > 0:
+            x0 = problem.get_reduced_vector(
+                result.optimize_result.list[0]['x']
+            )
+        # TODO multiple x0 for PT, #269
+
+    # set variational inference
+    variational_approx = PymcSampler()
+
+    # initialize sampler to problem
+    variational_approx.initialize(problem=problem, x0=x0)
+
+    # perform the sampling and track time
+    t_start = process_time()
+    variational_approx.fit(
+        n_iterations=n_iterations,
+        method=method,
+        random_seed=random_seed,
+        start_sigma=start_sigma,
+        **kwargs,
+    )
+    t_elapsed = process_time() - t_start
+    logger.info("Elapsed time: " + str(t_elapsed))
+
+    # extract results
+    # todo: build variational result object
+    return variational_approx.data
+
+    # if n_samples is not None:
+    #     vi_result.samples = variational_approx.data.sample(n_samples)
+    #
+    # # record time
+    # vi_result.time = t_elapsed
+    #
+    # # record results
+    # result.vi_result = vi_result
+    #
+    # autosave(
+    #     filename=filename,
+    #     result=result,
+    #     store_type="variational",
+    #     overwrite=overwrite,
+    # )
+    #
+    # return result
```

Comment:
Comments need to be removed and we need to build the variational result object as you said.